src/main/java/net/neoforged/moddevgradle/dsl/NeoForgeExtension.java

```diff
@@ -1,177 +1,50 @@
-     * The plugin automatically adds these dependencies to the main source set.
-     */
-    public void addModdingDependenciesTo(SourceSet sourceSet) {
-        var configurations = project.getConfigurations();
-        var sourceSets = ExtensionUtils.getSourceSets(project);
-        if (!sourceSets.contains(sourceSet)) {
-            throw new GradleException("Cannot add to the source set in another project.");
-        }
-
-        configurations.getByName(sourceSet.getRuntimeClasspathConfigurationName())
-                .extendsFrom(configurations.getByName(ModDevPlugin.CONFIGURATION_RUNTIME_DEPENDENCIES));
-        configurations.getByName(sourceSet.getCompileClasspathConfigurationName())
-                .extendsFrom(configurations.getByName(ModDevPlugin.CONFIGURATION_COMPILE_DEPENDENCIES));
-    }
-
-    /**
-     * NeoForge version number. You have to set either this, {@link #getNeoFormVersion()}
-     * or {@link #getNeoFormArtifact()}.
-     */
-    public abstract Property<String> getVersion();
-
-    /**
-     * You can set this property to a version of <a href="https://projects.neoforged.net/neoforged/neoform">NeoForm</a>
-     * to either override the version used in the version of NeoForge you set, or to compile against
-     * Vanilla artifacts that have no NeoForge code added.
-     * <p>
-     * This property is mutually exclusive with {@link #getNeoFormArtifact()}.
-     */
-    public abstract Property<String> getNeoFormVersion();
-
-    /**
-     * Is derived automatically from {@link #getVersion()}.
-     *
-     * @return Maven artifact coordinate (group:module:version)
-     */
-    public abstract Property<String> getNeoForgeArtifact();
-
-    /**
-     * Derived automatically from the {@link #getNeoFormVersion()}.
-     * You can override this property to use i.e. MCP for up to 1.20.1.
-     * <p>
-     * This property is mutually exclusive with {@link #getNeoForgeArtifact()}.
-     *
-     * @return Maven artifact coordinate (group:module:version)
-     */
-    public abstract Property<String> getNeoFormArtifact();
-
-    /**
-     * The list of additional access transformers that should be applied to the Minecraft source code.
-     * <p>
-     * If you do not set this property, the plugin will look for an access transformer file at
-     * {@code META-INF/accesstransformer.cfg} relative to your main source sets resource directories.
-     *
-     * @see <a href="https://projects.neoforged.net/neoforged/accesstransformers">Access Transformer File Format</a>
-     */
-    public void accessTransformers(Action<DataFileCollection> action) {
-        action.execute(accessTransformers);
-    }
-
-    public DataFileCollection getAccessTransformers() {
-        return accessTransformers;
+    @Deprecated(forRemoval = true)
+    public void setVersion(Object any) {
+        throw new InvalidUserCodeException("Please use enableModding { neoForgeVersion = ... } instead of the version property.");
```

Comment:
Please keep this and delegate it to `enableModding(s -> s.setNeoForgeVersion(any.toString()))`